src/libs/Navigation/NavigationRoot.js

```diff
@@ -43,15 +46,16 @@ class NavigationRoot extends Component {
             return;
         }
 
-        const path = getPathFromState(state, linkingConfig.config);
+        const currentPath = getPathFromState(state, linkingConfig.config);
 
         // Don't log the route transitions from OldDot because they contain authTokens
-        if (path.includes('/transition')) {
+        if (currentPath.includes('/transition')) {
             Log.info('Navigating from transition link from OldDot using short lived authToken');
         } else {
-            Log.info('Navigating to route', false, {path});
+            Log.info('Navigating to route', false, {path: currentPath});
         }
-        App.setCurrentURL(path);
+
+        this.setState({currentPath});
```

Comment:
NAB, I still feel icky about triggering a re-render of the `NavigationContainer` - but can't really back up this concern with any facts or evidence about why we should not do it so not gonna block on it.

Reply:
I see what you mean but for the record, child components only re-render if their props change. So it's safe to re-render the root component (or any super high ancestor) of an app so long as it doesn't cascade a massive change of props down the whole app.

Reply:
I think it doesn't matter much in this case because we are preventing re-renders in AuthScreens 😄